File: two_sum.java
class two_sum {
    public int[] twoSum(int[] nums, int target) {
        int arr[];
        int a=0;
        int b=0;
        int c=0;
        arr=new int[2];
        if(2<=nums.length && nums.length<= 10000)
        {
            for(int i=0;i<nums.length-1;i++)
            {
                for(int j=i+1;j<nums.length;j++)
                {
                    if(nums[i]+nums[j]==target)
                    {
                        a=i;
                        b=j;
                        c=1;
                        break;
                    }
                }
                if(c==1)
                    break;
            }
        }
        arr[0]=a;
        arr[1]=b;
        return arr;
    }
}

File: two_sum.py
# O(n) time | O(n) space
from typing import List
class two_sum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numMap = {}
        n = len(nums)
        # building hash map i.e. dictionary
        for i in range(n):
            complement = target - nums[i]
            if complement in numMap:
                return [numMap[complement], i]
            numMap[nums[i]] = i
